### DISCUSSION TITLE
Auth Cookie Setting

### DISCUSSION THREAD

**Discussion by Bcubed87** [awslabs/clickstream-web#58] [Q&A]
Should this field hold the value of the name of the auth cookie configured in AWS on the ALB  like this "**my-cookie-name**"or should the value contain a concatenation of the the name of the auth cookie and the value of the your oidc clients session cookie like this "**my-cookie-name**=**session-cookie-value**"?

**Comment by zxkane** [comment]
@Bcubed87 does [this example](https://github.com/aws-samples/alb-headless-authentication-client) code help?

**Reply by Bcubed87** [reply]
![image](https://github.com/user-attachments/assets/72c0cdb1-0be1-47c6-a7f6-7175f3ec7780)
vs
![image](https://github.com/user-attachments/assets/f3ae4f11-89c0-4d23-add5-1169bf77b1bd)

**Reply by dengmingtong** [reply]
The second one, like this "my-cookie-name=session-cookie-value"

**Reply by Bcubed87** [reply]
> The second one, like this "my-cookie-name=session-cookie-value"

That's great to know. When using this method, what I've noticed is that my cookie is not being sent with the request. The browser's tools shows no cookie sent even though I set the value. Looks like the fetch call made to the click analytics endpoint is dropping the cookies.

**Reply by zhu-xiaowei** [reply]
Hi～ @Bcubed87 sorry for late response, thanks for raising this problem, we are working on this issue in the past days.

Firstly, browsers will block all third-party cookies starting in 2025 by default. Currently, 1% of Chrome users have disabled third-party cookies. You can refer [cookie-countdown-chrome](https://developers.google.com/privacy-sandbox/blog/cookie-countdown-2024jan) and  [goodbye-third-party-cookies](https://developer.mozilla.org/en-US/blog/goodbye-third-party-cookies/) and check whether you are the 1% user.

Secondly, If you are using cross domain clickstream endpoint, we need to set the cross-domain authentication cookie through a new backend API which is the same domain with your clickstream endpoint, you can refer to [including_credentials](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#including_credentials) and [cross-domains-using-fetch-set-credentials](https://stackoverflow.com/questions/56965476/cors-cookie-not-set-on-cross-domains-using-fetch-set-credentials-include-an) and set the clickstream endpoint domain's cookie on your backend like: `Set-Cookie: my-cookie-name=session-cookie-value; SameSite=None; Secure; HttpOnly` before using clickstream SDK. After that the clickstream SDK will automatically added the cookie under header in the `/collect` request

In addition, you need to update clickstream web SDK to the latest version. At the same time, we have cached all the data that failed to be sent in localStorage. After the setting is successful, the failed data will be resent.
